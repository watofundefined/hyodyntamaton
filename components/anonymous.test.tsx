import React from 'react'
import { render, fireEvent } from 'test/test-utils'
import Anonymous from './anonymous'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Anonymous onLoginClicked={() => null} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Log in button', () => {
    const { getByText } = render(<Anonymous onLoginClicked={() => null} />)
    expect(() => getByText(/log in/i)).not.toThrow()
  })

  it('calls the onLoginClicked cb when clicked', () => {
    const cb = jest.fn()

    const { getByText } = render(<Anonymous onLoginClicked={cb} />)
    const loginButton = getByText(/log in/i)
    fireEvent.click(loginButton)

    expect(cb).toHaveBeenCalledTimes(1)
  })
})
