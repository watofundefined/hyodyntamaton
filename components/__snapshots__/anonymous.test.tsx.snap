// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anonymous component matches snapshot 1`] = `
<DocumentFragment>
  <header
    __self="[object Object]"
    __source="[object Object]"
  >
    <span
      __self="[object Object]"
      __source="[object Object]"
      class="logo"
    >
      Hyödyntämätön
    </span>
  </header>
  <main
    __self="[object Object]"
    __source="[object Object]"
  >
    App which translates beer reviews to your language.
    <button
      __self="[object Object]"
      __source="[object Object]"
      class="btn"
    >
      Log in with your Untappd account
    </button>
  </main>
</DocumentFragment>
`;
